File: 739-Daily-Temperatures/dailyTemperatures.java
class Solution {
    public int[] dailyTemperatures(int[] T) {
        Stack<Integer> stack = new Stack<Integer>();
        int[] ans = new int[T.length];
        
        for (int i = 0; i < T.length; i++) {
            while (!stack.isEmpty() && T[i] > T[stack.peek()]) {
                int pos = stack.pop();
                ans[pos] = i - pos;
            }
            stack.push(i);
        }
        
        return ans;
    }
}

File: 739-Daily-Temperatures/dailyTemperatures.js
var dailyTemperatures = function(T) {
    let stack = [];
    let ans = new Array(T.length).fill(0);
    
    for (let i = 0; i < T.length; i++) {
        while (stack.length && T[i] > T[stack[stack.length - 1]]) {
            let pos = stack.pop();
            ans[pos] = i - pos;
        }
        stack.push(i);
    }
    
    return ans;
};